commit 76abff3cfdad511f7d0bfe57c1a2b5913afc7b5b (refs/changes/42/2942/3)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2017-11-20T14:03:42-05:00 (1 year, 11 months ago)
    
    OS-6453 vmadm get core dump with custom zonename
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/src/vm/node_modules/vmload/index.js b/src/vm/node_modules/vmload/index.js
index 56496b1b..16476b2f 100644
--- a/src/vm/node_modules/vmload/index.js
+++ b/src/vm/node_modules/vmload/index.js
@@ -690,10 +690,10 @@ function getVmobj(uuid, options, callback)
             }
 
             if (wantField(options, 'last_modified')) {
-                if (cache.last_modified
+                if (cache.hasOwnProperty('last_modified')
                     && cache.last_modified.hasOwnProperty(vmobj.uuid)) {
 
-                    vmobj.last_modified = cache.last_modified[vmobj.zonename];
+                    vmobj.last_modified = cache.last_modified[vmobj.uuid];
                 } else {
                     log.warn('"last_modified" requested, but not found in cache'
                         + ' for VM ' + vmobj.uuid);
@@ -703,13 +703,13 @@ function getVmobj(uuid, options, callback)
             if (vmobj.zone_state !== 'running'
                     && wantField(options, 'exit_status')) {
 
-                if (cache.last_exited
+                if (cache.hasOwnProperty('last_exited')
                     && cache.last_exited.hasOwnProperty(vmobj.uuid)
                     && cache.last_exited[vmobj.uuid]
                     .hasOwnProperty('exit_status')) {
 
                     vmobj.exit_status
-                        = cache.last_exited[vmobj.zonename].exit_status;
+                        = cache.last_exited[vmobj.uuid].exit_status;
                 } else {
                     log.warn('"exit_status" requested, but not found in cache'
                         + ' for VM ' + vmobj.uuid);
@@ -719,13 +719,13 @@ function getVmobj(uuid, options, callback)
             if (vmobj.zone_state !== 'running'
                     && wantField(options, 'exit_timestamp')) {
 
-                if (cache.last_exited
+                if (cache.hasOwnProperty('last_exited')
                     && cache.last_exited.hasOwnProperty(vmobj.uuid)
                     && cache.last_exited[vmobj.uuid]
                     .hasOwnProperty('exit_timestamp')) {
 
                     vmobj.exit_timestamp
-                        = cache.last_exited[vmobj.zonename].exit_timestamp;
+                        = cache.last_exited[vmobj.uuid].exit_timestamp;
                 } else {
                     log.warn('"exit_timestamp" requested, but not found in '
                         + 'cache for VM ' + vmobj.uuid);
